src/components/YearlyDataTab.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from "react";
+import CopyButton from "./CopyButton";
+import { Button } from "library-simplified-reusable-components";
+import { LibraryData } from "../interfaces";
+
+export interface YearlyDataTabProps {
+  data: {[key: string]: LibraryData[]};
+}
+
+export interface YearlyDataTabState {
+  styled: boolean;
+}
+
+export default class YearlyDataTab extends React.Component<YearlyDataTabProps, YearlyDataTabState> {
+  private dataRef = React.createRef<HTMLUListElement>();
+  constructor(props: YearlyDataTabProps) {
+    super(props);
+    this.state = { styled: true };
+    this.toggleFormatting = this.toggleFormatting.bind(this);
+  }
+  sortByYear(data) {
+    let sortedByYear = {};
+    Object.keys(data).map((name: string) => {
+      let category = data[name];
+      category.forEach((library) => {
+        let year = library.basic_info.timestamp && library.basic_info.timestamp.match(/20\d+/)[0];
+        if (year) {
+          if (sortedByYear[year]) {
+            sortedByYear[year][name].push(library.basic_info.name);
```

Comment:
I'm a bit confused. What is the first `name` in this case?

Reply:
Category name ("production", "testing", or "cancelled")

Reply:
Unless if the first name is either production, testing, or cancelled? Maybe a better name might clear things up.